pallets/elections/src/impls.rs

```diff
@@ -0,0 +1,316 @@
+                    blocks_per_session,
+                    &validator.into(),
+                    total,
+                ),
+            )
+        });
+
+        pallet_staking::Pallet::<T>::reward_by_ids(session_rewards);
+    }
+
+    // Choose a subset of all the validators for current era that contains all the
+    // reserved nodes. Non reserved ones are chosen in consecutive batches for every session
+    fn rotate_committee() -> Option<Vec<T::AccountId>> {
+        let current_era = match pallet_staking::ActiveEra::<T>::get() {
+            Some(ae) if ae.index > 0 => ae.index,
+            _ => return None,
+        };
+        let all_validators: Vec<T::AccountId> =
+            pallet_staking::ErasStakers::<T>::iter_key_prefix(current_era).collect();
+        let reserved = ErasReserved::<T>::get();
+        let n_validators = MembersPerSession::<T>::get() as usize;
+        let current_session = pallet_session::Pallet::<T>::current_index();
+
+        rotate(
+            current_era,
+            current_session,
+            n_validators,
+            all_validators,
+            reserved,
+        )
+    }
+
+    fn populate_reserved_on_next_era_start(start_index: SessionIndex) {
+        let current_era = match pallet_staking::ActiveEra::<T>::get() {
+            Some(ae) => ae.index,
+            _ => return,
+        };
+        // this will be populated once for the session `n+1` on the start of the session `n` where session
+        // `n+1` starts a new era.
+        if let Some(era_index) = pallet_staking::ErasStartSessionIndex::<T>::get(current_era + 1) {
+            if era_index == start_index {
+                let reserved_validators = pallet_staking::Invulnerables::<T>::get();
+                ErasReserved::<T>::put(reserved_validators);
+            }
+        }
+    }
+
+    fn adjust_rewards_for_session() {
+        match pallet_staking::ActiveEra::<T>::get() {
+            Some(ae) if ae.index == 0 => return,
+            None => return,
+            _ => {}
+        }
+
+        let committee = pallet_session::Validators::<T>::get();
+        Self::reward_for_session_non_committee(&committee);
+    }
+}
+
+impl<T> pallet_authorship::EventHandler<T::AccountId, T::BlockNumber> for Pallet<T>
+where
+T: Config + pallet_session::Config + pallet_staking::Config,
+<<T as pallet_staking::Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance: Into<u128>,
+<T as pallet_session::Config>::ValidatorId: From<T::AccountId> {
+    fn note_author(validator: T::AccountId) {
+        let active_era = pallet_staking::ActiveEra::<T>::get().unwrap().index;
+
+        let total = Self::scaled_total_exposure(active_era, &validator);
+        let sessions_per_era = T::SessionsPerEra::get();
+
+        let validator_points_per_block =
+            validator_points_per_block(total, sessions_per_era, Self::blocks_to_produce_per_session());
+
+        pallet_staking::Pallet::<T>::reward_by_ids(
```

Comment:
I believe that 1. is not much of a problem really but point 3 is enough to justify new storage and I agree that performance hit should be marginal. The only downside I see is that we need to change UI so that at least it shows an info that we update points every 15mins. This however has low prio in my opinion.

Reply:
In the new version the points will be actually hard to interpret (as they drip proportionally to stake), so it's better if people just ignore them. Ideally we should remove points from the UI, but that's probably too much work. Anyway, one column with random noise that changes every 15 minutes have never hurt anyone :P

Reply:
In my opinion, we can hide the `points` column from users entirely. There's another column just right to it `last #` which shows the progress that validators are producing blocks. We can treat (and announce) `points` as an internal metric, which will directly translate into rewards, but those are the internals. One can still query them from storage if needed. With the lenient uptime feature as noted above some users might notice some incontinence at the end of an era when points are adjusted. 

![image](https://user-images.githubusercontent.com/3909333/168983621-42b3b419-5b7f-44c7-9590-690d7c06852a.png)

Reply:
not related to ui: the code change is in place :)

Reply:
jira issue for it is here: [A0-955](https://cardinal-cryptography.atlassian.net/browse/A0-955)